src/main/java/com/moodysalem/phantomjs/wrapper/PhantomJSSetup.java

```diff
@@ -0,0 +1,173 @@
+package com.moodysalem.phantomjs.wrapper;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import com.moodysalem.phantomjs.wrapper.beans.OperatingSystem;
+
+public class PhantomJSSetup{
+	
+	protected final static Logger logger = Logger.getLogger(PhantomJSSetup.class.getName());
+	
+	// this will store a reference to the executable phantomjs binary after we unzip the resource
+    protected static File PHANTOM_JS_BINARY = null;
```

Comment:
We should make this final and initialize it in a static block for this class-versus initializing it in the PhantomJS class